src/__tests__/host-component-names.test.tsx

```diff
@@ -11,10 +16,41 @@ const mockGetQueriesForElements = jest.spyOn(
 ) as jest.Mock;
 
 describe('getHostComponentNames', () => {
+  test('returns host component names from internal config', () => {
+    configureInternal({
+      hostComponentNames: { text: 'banana', textInput: 'banana' },
+    });
+
+    expect(getHostComponentNames()).toEqual({
+      text: 'banana',
+      textInput: 'banana',
+    });
+  });
+
+  test('throws when names are missing in the internal config', () => {
+    expect(() => getHostComponentNames()).toThrowErrorMatchingInlineSnapshot(`
```

Comment:
I think the error message for `getHostComponentNames` function should be updated, it will throw if `configureHostComponentNamesIfNeeded` has not been called before but that may not be linked to the configuration and would likely be a bug on our end, wdyt ?

Reply:
Resolved by triggering lazy detection inside `getHostComponentNames`